src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -1444,15 +1444,15 @@ private IEnumerable<CommandInfo> GetMatchingCommandsFromModules(string commandNa
                             // Look in function table
                             if ((this.CommandType & (CommandTypes.Function | CommandTypes.Filter | CommandTypes.Configuration)) != 0)
                             {
-                                foreach (DictionaryEntry function in module.SessionState.Internal.GetFunctionTable())
+                                foreach (var function in module.SessionState.Internal.GetFunctionTable())
```

Comment:
The same.

Reply:
```csharp
SessionState moduleSessionState = module.SessionState;
SessionStateInternal sessionStateInternal = moduleSessionState.Internal;
IDictionary<string, FunctionInfo> functionTable = sessionStateInternal.GetFunctionTable();
foreach (var function in functionTable)
{
...
```
?

Reply:
Just a few lines below, we have:
```csharp
if (func.Module.Path.Equals(module.Path, StringComparison.OrdinalIgnoreCase))
    yield return function.Value;
```

That must obviously be rewritten as
```csharp
PSModuleInfo psModuleInfo = func.Module;
string path = psModuleInfo.Path;
string modulePath = module.Path;
if (path.Equals(modulePath, StringComparison.OrdinalIgnoreCase))
{
    FunctionInfo matchingCommandsFromModules = function.Value;
    yield return matchingCommandsFromModules;
}
```
otherwise the types are not obvious.

Reply:
I suppose we should not delve into refactoring until this helps follow nullable annotating.